In [1]:
import os
import torch
import pandas as pd
import optuna

from definitions import *

In [2]:
optim_path = os.path.join(EXP_DIR_PATH, 'bi-lstm', 'optimization')
# study_name = 'bi-lstm_wAtt_sTPE_pNone_df0.2_wf0.2.pkl'
# files_n_params = [sn for sn in optim_path if 'featOptimSentFeat' in sn.split('_')[1]]


In [10]:
optim_type = 'featOptimSentFeat'
# optim_type = 'featOptimFeatModel'
# optim_type = 'featOptimWordLevel'
# TODO: store results in pd.DataFrame to easily use with plotly, create awesome charts
for feature_type in ['pos', 'tag', 'dep']:
    file_name = f'bi-lstm_{optim_type}_{feature_type}_mF1_wAtt_sGrid_pNone_df0.2_wf0.03'
    # file_name = f'bi-lstm_{optim_type}_mF1_wAtt_sGrid_pNone_df0.2_wf0.03'
    
    study_path = os.path.join(optim_path, f'{file_name}.pkl')
    params_path = os.path.join(optim_path, f'{file_name}_params.pkl')
    study = torch.load(study_path)
    params = torch.load(params_path)

    print(study.best_params)
    print(params)

{'stopwords_type': 'wstop', 'from_selection': True, 'pos_feature_type': 'none', 'word_count_feature_type': 'count_words'}
{'batch_size': 32, 'pooling_strategy': 'last_four_sum', 'dropout': 0.01, 'hidden_dim': 128, 'with_sequential_layer': False, 'learning_rate': 4.8949721500587447e-05, 'optimizer_weigth_decay': 1.0859798578698863e-05, 'pos_weight': 32.81159420289855}
{'stopwords_type': 'wstop', 'from_selection': False, 'tag_feature_type': 'none', 'word_count_feature_type': 'count_words'}
{'batch_size': 32, 'pooling_strategy': 'last_four_sum', 'dropout': 0.01, 'hidden_dim': 128, 'with_sequential_layer': False, 'learning_rate': 4.8949721500587447e-05, 'optimizer_weigth_decay': 1.0859798578698863e-05, 'pos_weight': 32.81159420289855}
{'stopwords_type': 'wostop', 'from_selection': True, 'dep_feature_type': 'onehot', 'word_count_feature_type': 'count_words'}
{'batch_size': 16, 'pooling_strategy': 'last_four', 'dropout': 0.05, 'hidden_dim': 128, 'with_sequential_layer': False, 'learning_rate

**bi-lstm feature optim**
- params: {
    'batch_size': 16, 'pooling_strategy': 'last_four', 'dropout': 0.05, 'hidden_dim': 128, 'with_sequential_layer': False, 'learning_rate': 1.0691418898885528e-05, 'optimizer_weigth_decay': 1.0929364050970057e-05, 'pos_weight': 32.81159420289855, 'fnn_hidden_dim': 256, 'fnn_n_hidden_layers': 4, 'fnn_dropout': 0.12, 'pnn_hidden_dim': 8, 'pnn_dropout': 0.24
}

- featOptimSentFeat:
    - pos: {
        'stopwords_type': 'wstop', 'from_selection': True, 'pos_feature_type': 'none', 'word_count_feature_type': 'count_words'
    }
    - tag: {
        'stopwords_type': 'wstop', 'from_selection': False, 'tag_feature_type': 'none', 'word_count_feature_type': 'count_words'
    }
    - dep: {
        'stopwords_type': 'wostop', 'from_selection': True, 'dep_feature_type': 'onehot', 'word_count_feature_type': 'count_words'
    }

- featOptimFeatModel:
    - pos: {
        'stopwords_type': 'wstop', 'from_selection': False, 'pos_feature_type': 'sum', 'word_count_feature_type': 'none'
    }
    - tag: {
        'stopwords_type': 'wstop', 'from_selection': False, 'tag_feature_type': 'sum', 'word_count_feature_type': 'none'
    }
    - dep: {
        'stopwords_type': 'wstop', 'from_selection': True, 'dep_feature_type': 'none', 'word_count_feature_type': 'none'
    }

- featOptimWordLevel
    - {
        'stopwords_type': 'wstop', 'word_count_feature_type': 'count_words', 'word_level_feature_type': 'dep'
    }

In [10]:
import time
f = ['a', 'b', 'c']
for x in f:
    
    print('aktualne pismeno: ', x, end='\r')
    time.sleep(3)

KeyboardInterrupt: 